src/sources/soundsourceproviderregistry.h

```diff
@@ -4,54 +4,109 @@
 
 #include <QMap>
-#include <QRegExp>
 
 namespace Mixxx {
 
+class SoundSourceProviderRegistration {
+public:
+    const SoundSourcePluginLibraryPointer& getPluginLibrary() const {
+        return m_pPluginLibrary;
+    }
+    const SoundSourceProviderPointer& getProvider() const {
+        return m_pProvider;
+    }
+    SoundSourceProviderPriority getProviderPriority() const {
+        return m_providerPriority;
+    }
+
+private:
+    friend class SoundSourceProviderRegistry;
+    SoundSourceProviderRegistration(
+            SoundSourcePluginLibraryPointer pPluginLibrary,
+            SoundSourceProviderPointer pProvider,
+            SoundSourceProviderPriority providerPriority)
+        : m_pPluginLibrary(pPluginLibrary),
+          m_pProvider(pProvider),
+          m_providerPriority(providerPriority) {
+    }
+
+    SoundSourcePluginLibraryPointer m_pPluginLibrary;
+    SoundSourceProviderPointer m_pProvider;
+    SoundSourceProviderPriority m_providerPriority;
+};
+
 // Registry for SoundSourceProviders
 class SoundSourceProviderRegistry {
 public:
-    SoundSourceProviderPointer registerProvider(
+    // Registers a provider for all supported file extensions
+    // with their implicit priority hint.
+    void registerProvider(
             const SoundSourceProviderPointer& pProvider);
-    SoundSourceProviderPointer registerPluginLibrary(
+    // Registers a provider from a plugin library for all supported
+    // file extensions with their implicit priority hint.
+    void registerPluginLibrary(
             const SoundSourcePluginLibraryPointer& pPluginLibrary);
 
-    // Completes the registration by building the corresponding
-    // regular expressions for file names.
-    void finishRegistration();
-
-    SoundSourceProviderPointer getProviderForFileExtension(
-            const QString& fileExtension) const {
-        return m_entries.value(fileExtension).pProvider;
-    }
+    // Registers a provider for a single file extension with an
+    // explicitly specified priority. The provider must support
+    // the given file extension.
+    void registerProviderForFileExtension(
+            const QString& fileExtension,
+            const SoundSourceProviderPointer& pProvider,
+            SoundSourceProviderPriority providerPriority);
+    // Registers a provider from a plugin library for a single file
+    // extension with an explicitly specified priority. The provider
+    // must support the given file extension.
+    void registerPluginProviderForFileExtension(
+            const QString& fileExtension,
+            const SoundSourcePluginLibraryPointer& pPluginLibrary,
+            const SoundSourceProviderPointer& pProvider,
+            SoundSourceProviderPriority providerPriority);
+
+    // Reregister a provider that is already registered with
+    // a new priority. It doesn't matter if the provider is
+    // built-in or provided by a plugin library.
+    void reregisterProviderForFileExtension(
```

Comment:
This function seams to be unused. 
It should be called like: changeProviderPriorityForFileExtension()